setup.sh

```diff
@@ -68,33 +92,32 @@ function handleCmdLine() {
 #
 function confirmChoices() {
     function joinWithWorkingDir() {
-        for i in $*; do
+        for i in "$@"; do
             echo "$wd/$i"
         done
     }
 
-    for project in $projects; do
-        if [ -e "$wd/$project" ]; then
-            toUpdate=($toUpdate "$project")
+    for project in "${projects[@]}" ; do
+        if [ -d "$wd/$project" ] ; then
+            toUpdate+=("$project")
         else
-            toInstall=($toInstall "$project")
+            toInstall+=("$project")
         fi
     done
-    if [[ ! -z $toInstall ]]; then
+    if [[ ${#toInstall[@]} -gt 0 ]]; then
         echo "*** The following projects will be INSTALLED:"
-        joinWithWorkingDir ${toInstall}
-        echo "*** Note: this script assumes you have a github account set up."
+        joinWithWorkingDir "${toInstall[@]}"
     fi
-    if [[ ! -z $toUpdate ]]; then
+    if [[ ${#toUpdate[@]} -gt 0 ]]; then
         echo "*** The following projects will be UPDATED:"
-        joinWithWorkingDir ${toUpdate}
+        joinWithWorkingDir "${toUpdate[@]}"
     fi
 
-    echo "Is this what you want?"
+    echo "Is this what you want? [y|n]"
     local yn
     while true; do
-        read yn
-        case $yn in
+        read -r yn
+        case "$yn" in
             [Yy]* ) break;;
             [Nn]* ) exit;;
             * ) echo "Please answer y or n.";;
```

Comment:
Typically the behavior is to ask "Y|n" and accept an empty new line as "yes", but I didn't want to put big changes into this PR.